x-pack/plugins/advanced_ui_actions/public/components/action_wizard/action_wizard.tsx

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState } from 'react';
+import {
+  EuiButtonEmpty,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiIcon,
+  EuiSpacer,
+  EuiText,
+  EuiKeyPadMenuItem,
+} from '@elastic/eui';
+import { txtChangeButton } from './i18n';
+import './action_wizard.scss';
+
+// TODO: this interface is temporary for just moving forward with the component
+// and it will be imported from the ../ui_actions when implemented properly
+// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
+export type ActionFactoryBaseConfig = {};
+export interface ActionFactory<Config extends ActionFactoryBaseConfig> {
+  type: string;
+  displayName: string;
+  iconType?: string;
+  wizard: React.FC<ActionFactoryWizardProps<Config>>;
+}
+
+export interface ActionFactoryWizardProps<Config extends ActionFactoryBaseConfig> {
+  /**
+   * Current (latest) config of the item. (state)
+   */
+  config: Config | null;
```

Comment:
Thoughts on using `| undefined` instead of `| null`?

Reply:
`Config | null` idea is mostly for `onConfig` callback. when config is emitted - it means that user's input is valid / complete. When `null` - it means either input is invalid or incomplete 
https://user-images.githubusercontent.com/7784120/75778113-7c780600-5d57-11ea-8e04-a319d046d02b.png

This `config` is actually becoming initialConfig? in follow up pr. and need to support editing cases.

Reply:
Went with `factory.isValid(config)` approach